pandas/tests/io/test_sql.py

```diff
@@ -1221,470 +1246,695 @@ class DummyException(Exception):
-        result = sql.to_sql(temp_frame, "test_index_label", self.conn)
-        assert result == expected_row_count
-        frame = sql.read_sql_query("SELECT * FROM test_index_label", self.conn)
-        assert frame.columns[0] == "level_0"
-        assert frame.columns[1] == "level_1"
+    df = to_timedelta(Series(["00:00:01", "00:00:03"], name="foo")).to_frame()
+    with tm.assert_produces_warning(UserWarning):
+        result_count = df.to_sql(name="test_timedelta", con=conn)
+    assert result_count == 2
+    result = sql.read_sql_query("SELECT * FROM test_timedelta", conn)
+    tm.assert_series_equal(result["foo"], df["foo"].view("int64"))
+
+
+@pytest.mark.db
+@pytest.mark.parametrize("conn", all_connectable)
+def test_api_complex_raises(conn, request):
+    conn = request.getfixturevalue(conn)
+    df = DataFrame({"a": [1 + 1j, 2j]})
+    msg = "Complex datatypes not supported"
+    with pytest.raises(ValueError, match=msg):
+        assert df.to_sql("test_complex", con=conn) is None
 
-        # specifying index_label
-        result = sql.to_sql(
-            temp_frame,
-            "test_index_label",
-            self.conn,
-            if_exists="replace",
-            index_label=["A", "B"],
-        )
-        assert result == expected_row_count
-        frame = sql.read_sql_query("SELECT * FROM test_index_label", self.conn)
-        assert frame.columns[:2].tolist() == ["A", "B"]
 
+@pytest.mark.db
+@pytest.mark.parametrize("conn", all_connectable)
+@pytest.mark.parametrize(
+    "index_name,index_label,expected",
+    [
+        # no index name, defaults to 'index'
+        (None, None, "index"),
+        # specifying index_label
+        (None, "other_label", "other_label"),
         # using the index name
-        temp_frame.index.names = ["A", "B"]
-        result = sql.to_sql(
-            temp_frame, "test_index_label", self.conn, if_exists="replace"
+        ("index_name", None, "index_name"),
+        # has index name, but specifying index_label
+        ("index_name", "other_label", "other_label"),
+        # index name is integer
+        (0, None, "0"),
+        # index name is None but index label is integer
+        (None, 0, "0"),
+    ],
+)
+def test_api_to_sql_index_label(conn, request, index_name, index_label, expected):
+    conn = request.getfixturevalue(conn)
+    if sql.has_table("test_index_label", conn):
+        with sql.SQLDatabase(conn, need_transaction=True) as pandasSQL:
+            pandasSQL.drop_table("test_index_label")
+
+    temp_frame = DataFrame({"col1": range(4)})
+    temp_frame.index.name = index_name
+    query = "SELECT * FROM test_index_label"
+    sql.to_sql(temp_frame, "test_index_label", conn, index_label=index_label)
+    frame = sql.read_sql_query(query, conn)
+    assert frame.columns[0] == expected
+
+
+@pytest.mark.db
+@pytest.mark.parametrize("conn", all_connectable)
+def test_api_to_sql_index_label_multiindex(conn, request):
+    conn_name = conn
+    if "mysql" in conn_name:
```

Comment:
I wasn't able to reproduce this locally using MariaDB but kept seeing it on CI. Not sure if it is a version thing or MySQL <> MariaDB difference.